### PR TITLE
SPU2-X: Minor GUI enhancements, remove About Box.

### PR DESCRIPTION
**Pull request by MrCK1**
This PR disables some buttons/dropdowns in the GUI for cases when they can't be used. I also removed the unused remaining elements of the About Box that were leftover.

### CODE REVIEW

**Comment by MrCK1** [comment]
Any reason as to why clang-format is failing in Travis? I had a feeling some tabs/spaces got mixed in or something but I have no idea at this point 😛

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 1]
```diff
@@ -261,6 +261,17 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
             SendDialogMsg(hWnd, IDC_VOLUME_SLIDER, TBM_SETPOS, TRUE, configvol);
             swprintf_s(temp, L"%d%%", configvol);
             SetWindowText(GetDlgItem(hWnd, IDC_VOLUME_LABEL), temp);
+			
+			const int module = (int)SendMessage(GetDlgItem(hWnd, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
```
Comment (danilaml) [inline plugins/spu2-x/src/Windows/Config.cpp]: tabs

Comment (gregory38) [inline plugins/spu2-x/src/Windows/Config.cpp]: the idea is that you ask clang to reformat and don't even bother to know what are the issues :)

File: plugins/spu2-x/Website.txt [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-	http://code.google.com/p/pcsx2/
-
-	You'll find information on how to check out from svn and how to compile
-	Pcsx2 and the plugins at the above link.
-
-SPU2-X (svn):
-	http://pcsx2.googlecode.com/svn/trunk/plugins/spu2-x
-
-Pcsx2 + All Plugins (svn):
-	http://pcsx2.googlecode.com/svn/trunk/
```
Comment (turtleli) [inline plugins/spu2-x/Website.txt]: This file has already been removed on the master branch.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 3]
```diff
@@ -294,6 +305,19 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
                     EndDialog(hWnd, 0);
                     break;
 
+				case IDC_OUTPUT:
+					if (wmEvent == CBN_SELCHANGE) {
+						const int module = (int)SendMessage(GetDlgItem(hWnd, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
+						bool IsConfigurable = module != 0;
+						bool XAudio2Expansion = module == 1;
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: IIRC Audio expansion also works on PortAudio.

File: plugins/spu2-x/src/Windows/AboutBox.cpp [hunk 4]
```diff
@@ -1,66 +0,0 @@
...
-        default:
-            return FALSE;
-    }
-    return TRUE;
-}
-
-void AboutBox()
-{
-    DialogBox(hInstance, MAKEINTRESOURCE(IDD_ABOUT), GetActiveWindow(), (DLGPROC)AboutProc);
-}
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/AboutBox.cpp]: IMO the about box code removal should be in a separate commit.

**Comment by MrCK1** [comment]
@turtleli Everything asked was done. I still have to fix clang _again_.

I also noted a while ago was that the XAudio2 Configuration dialog doesn't work. Can you maybe look into that, I don't know if it's related to XAudio2 being 2 separate modules.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 5]
```diff
@@ -294,6 +305,19 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
                     EndDialog(hWnd, 0);
                     break;
 
+                case IDC_OUTPUT:
+                    if (wmEvent == CBN_SELCHANGE) {
+                    const int module = (int)SendMessage(GetDlgItem(hWnd, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
```
Comment (danilaml) [inline plugins/spu2-x/src/Windows/Config.cpp]: Why this If is so weirdly formatted? Is this case label perhaps missing a `{`, `}`?

Comment (MrCK1) [inline plugins/spu2-x/src/Windows/Config.cpp]: I don't see anything wrong with the formatting (that I know of) but i'll check in a bit when I'm not on a mobile screen.

Comment (danilaml) [inline plugins/spu2-x/src/Windows/Config.cpp]: if and it's contents have the same indentation level:
```cpp
                    if (wmEvent == CBN_SELCHANGE) {
                    const int module = (int)SendMessage(GetDlgItem(hWnd, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
```
And `if` brace ends at `} break;`, while looking at the next case it seems like the entire case body (including `if`) was supposed to be inside a case block.

File: plugins/spu2-x/src/Config.h [hunk 6]
```diff
@@ -99,5 +99,4 @@ extern void ApplySettings(soundtouch::SoundTouch &sndtouch);
 
 extern void ReadSettings();
 extern void WriteSettings();
-extern void configure();
-extern void AboutBox();
+extern void configure();
```
Comment (turtleli) [inline plugins/spu2-x/src/Config.h]: I'd prefer keeping the newline (my Linux text editor (vim) will re-introduce one if I modify the file).

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 7]
```diff
@@ -294,6 +305,19 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+                    bool IsConfigurable = module != 0;
+                    bool XAudio2Expansion = module == 1 || module == 4;
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: Badly named variable since it's not just XAudio2 expansion.

Also, hardcoded values aren't great.

**Comment by turtleli** [comment]
> I still have to fix clang again.

The idea is that you're supposed to run clang-format. There's a VS plugin, or you can use the command line if you have LLVM stuff installed.

> I also noted a while ago was that the XAudio2 Configuration dialog doesn't work. Can you maybe look into that, I don't know if it's related to XAudio2 being 2 separate modules.

There's not meant to be one because there's nothing to configure.

On the About box removal - it would be nice to remove the Linux About box too. It's up to you whether you do it though.

**Comment by MrCK1** [comment]
I can remove the one on Linux as well. I just reinstalled win 10 and VS 2015, so I'm all set to go.

I'll take your word for it on the XAudio dialog, but then I don't know why there's an unused entry in the .rc file.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 7 above)
Comment (MrCK1) [inline plugins/spu2-x/src/Windows/Config.cpp]: I'll change the car name. Can you explain what's "wrong" with the hardcoded values? - I don't know which line you're referring to :)

Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: Well, what does `1` and `4` refer to? They may currently refer to XAudio2 and PortAudio, but if, for instance, someone removes the DirectSound or WaveOut module or add a new Windows module, then that'll likely change.

**Comment by turtleli** [comment]
> I'll take your word for it on the XAudio dialog, but then I don't know why there's an unused entry in the .rc file

Well, you shouldn't take my word for it ;) I don't know if the XAudio2 dialog was ever used.

**Comment by MrCK1** [comment]
@turtleli Changes are a WIP, as I'm still finding time to finish them. 

Would you mind if I added another commit to remove DirectSound in this PR as well, or should I save it for later? (I'll make the appropriate changes to the hardcoded values as you mentioned earlier) I had been thinking about it for a while anyways, and I'm not sure if WaveOut is still useful or not.

**Comment by turtleli** [comment]
> Would you mind if I added another commit to remove DirectSound in this PR as well, or should I save it for later?

I'd prefer you not add it to this PR.

Also, I think you should provide some follow up to your other PRs.

**Comment by gregory38** [comment]
@MrCK1 Please read https://github.com/PCSX2/pcsx2/wiki/ClangFormat-Visual-Studio-plug-in

**Comment by MrCK1** [comment]
@turtleli Everything's all set here. I'll squash this in a little bit.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 8]
```diff
@@ -262,6 +262,17 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
             swprintf_s(temp, L"%d%%", configvol);
             SetWindowText(GetDlgItem(hWnd, IDC_VOLUME_LABEL), temp);
 
+            const int module = (int)SendMessage(GetDlgItem(hWnd, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
+            bool IsConfigurable = module != 0;
+            bool XAudio2Expansion = module == 1 || module == 4;
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: Variable name and hardcoded values haven't been resolved (https://github.com/PCSX2/pcsx2/pull/1963#discussion_r122554260).

Also could use `const bool`.

File: plugins/spu2-x/src/CMakeLists.txt [hunk 9]
```diff
@@ -88,7 +88,6 @@ if(Windows)
     include_directories("Windows")
 else()
     LIST(APPEND spu2xSources
-        Linux/AboutBox.cpp
```
Comment (turtleli) [inline plugins/spu2-x/src/CMakeLists.txt]: It'd be a good idea to also remove Linux/AboutBox.cpp from the repo.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 10]
```diff
@@ -262,6 +262,17 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+
+            EnableWindow(GetDlgItem(hWnd, IDC_OUTCONF), IsConfigurable);
+            EnableWindow(GetDlgItem(hWnd, IDC_VOLUME_SLIDER), IsConfigurable);
+            EnableWindow(GetDlgItem(hWnd, IDC_LATENCY_SLIDER), IsConfigurable);
+            EnableWindow(GetDlgItem(hWnd, IDC_VOLUME_SLIDER), IsConfigurable);
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: Duplicate IDC_VOLUME_SLIDER line.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 11]
```diff
@@ -294,6 +305,20 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+                    if (wmEvent == CBN_SELCHANGE) {
+                        const int module = (int)SendMessage(GetDlgItem(hWnd, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
+                        bool IsConfigurable = module != 0;
+                        bool XAudio2Expansion = module == 1 || module == 4;
+
+                        EnableWindow(GetDlgItem(hWnd, IDC_OUTCONF), IsConfigurable);
+                        EnableWindow(GetDlgItem(hWnd, IDC_VOLUME_SLIDER), IsConfigurable);
+                        EnableWindow(GetDlgItem(hWnd, IDC_LATENCY_SLIDER), IsConfigurable);
+
+                        EnableWindow(GetDlgItem(hWnd, IDC_SPEAKERS), XAudio2Expansion);
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: This looks like a copy of the above code. It might be a good idea to put the common stuff into a function.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 8 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: It looks like I'll have to use constant bool because It doesn't look like I can convert the values
 values any other way. 

```const bool XAudio2Expansion = module == 1 || module == 4;```
So would this be O.K. besides the variable name? (I'll change that later)

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 10 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: Done.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 11 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: You are correct, it is a copy of the code above because the proper values need to be applied during init of the dialog and final output. Can you explain more about putting this into a function? It's already contained in a ```case WM_INITDIALOG``` and ```case IDC_OUTPUT``` in both instances.

File: plugins/spu2-x/src/CMakeLists.txt (hunk 9 above)
Comment (MrCK1) [reply plugins/spu2-x/src/CMakeLists.txt]: I can't find it in my repo, I thought you removed that a while ago?

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 11 above)
Comment (turtleli) [reply plugins/spu2-x/src/Windows/Config.cpp]: `case` isn't a function, it's a label.

By put in function I mean create a function like:
```
void common_code_function_please_name_well(HWND window)
{
    // common code here
}
```
and then call that function instead of having duplicated code.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 8 above)
Comment (turtleli) [reply plugins/spu2-x/src/Windows/Config.cpp]: I copy my previous response:
> Well, what does `1` and `4` refer to? They may currently refer to XAudio2 and PortAudio, but if, for instance, someone removes the DirectSound or WaveOut module or add a new Windows module, then that'll likely change.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 12]
```diff
@@ -262,6 +262,17 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
             swprintf_s(temp, L"%d%%", configvol);
             SetWindowText(GetDlgItem(hWnd, IDC_VOLUME_LABEL), temp);
 
+            const int module = (int)SendMessage(GetDlgItem(hWnd, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
```
Comment (ssakash) [inline plugins/spu2-x/src/Windows/Config.cpp]: Is an integer cast necessary here? (I don't think it's needed as ``CB_GETCURSEL`` returns the value of the index associated with the object, which already has an int data type)

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 8 above)
Comment (ssakash) [reply plugins/spu2-x/src/Windows/Config.cpp]: For handling hard coded values - I'd prefer if enumerations are use over macros.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 11 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: That's what I figured it was lol. I'll do some work on it when I get a chance in a bit.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 8 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: @turtleli I'm still stuck on how to get rid of the hardcoded values. This works, but I can't use logical OR with it and I'm pretty sure it's the same as before.

```
	const int OutputModule = (int)SendMessage(GetDlgItem(window, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
	const bool AudioExpansion = mods[OutputModule] != XAudio2Out;
	const bool IsConfigurable = mods[OutputModule] == XAudio2Out || PortaudioOut || WaveOut || DSoundOut;
```

Comment (willkuer) [reply plugins/spu2-x/src/Windows/Config.cpp]: This is wrong on avery basic level. '==' binds stronger than '||'. So the first part of your conditoonal chain will be either true or false and afterwards you OR them with PortAudioOut, WaveOut and so on which are numbers. Only combine true or false with AND and OR...

What you really want to do is check whether the module is xaudio or the module is portaudio or the module is waveout and so on.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 12 above)
Comment (willkuer) [reply plugins/spu2-x/src/Windows/Config.cpp]: Actually cast to an enum class would be reasonable here. (But also using the same enum class as datatype for module).

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 8 above)
Comment (turtleli) [reply plugins/spu2-x/src/Windows/Config.cpp]: Yeah, as willkuer said, you've made a very basic mistake.
```
const bool IsConfigurable = mods[OutputModule] == XAudio2Out || PortaudioOut || WaveOut || DSoundOut;
```
is essentially the same as:
```
const bool IsConfigurable = (mods[OutputModule] == XAudio2Out) || (PortaudioOut != nullptr)
    || (WaveOut != nullptr) || (DSoundOut != nullptr);
```
which unconditionally evaluates to true, and I kinda doubt that's intended.

Note that `XAudio2Out` only covers Windows 8+. `XAudio2_27_Out` covers earlier Windows versions.

Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: OK, so I assume this method can't be used at all right?

On a side note, I already have this working inside a function. I'm also going to see if it's worth putting some of the slider stuff into a function as well (it seems a little copy and paste) but If I can't figure out a way to change these hardcoded values; it may be best that you take over the rest of the PR ;)

Comment (willkuer) [reply plugins/spu2-x/src/Windows/Config.cpp]: Disclaimer: What I wrote before about these DSoundOut, WaveOut, ... being numbers was wrong. I thought they are enums as I didn't read the code. As turtleli pointed out they are pointers (of type SndOutModule*) so you don't OR numbers but pointers with the first initial bool (which leads to the line turtleli posted). What you however want to do is OR bools. Each bool checks whether the current module is pointing to the DSoundOut, WaveOut, ... module.


Instead of 
```
const bool IsConfigurable =
    (mods[OutputModule] == XAudio2Out) ||
    (PortaudioOut != nullptr) ||
    (WaveOut != nullptr) ||
    (DSoundOut != nullptr);
```

you want to know

```
const bool IsConfigurable =
    (mods[OutputModule] == XAudio2Out) ||
    (mods[OutputModule] == PortaudioOut) ||
    (mods[OutputModule] == WaveOut) ||
    (mods[OutputModule] == DSoundOut);
```

Btw. this is a really nice example why OOP is a good concept for module programming. This could be handled in appropriate child classes without looking that wonky.

Comment (turtleli) [reply plugins/spu2-x/src/Windows/Config.cpp]: @MrCK1 Some things I forgot to mention in my previous comment - When you check whether it's configurable or not, I don't get why you don't just check that it's not the null output module. It's what you initially did as well. And for audio expansion, you seem to have omitted PortAudio again. As for fixing the logic, just read what willkuer posted and adapt.

**Comment by MrCK1** [comment]
@turtleli The changes are 99% done, I'm just trying to fix one of the buttons that doesn't enable properly. 


Anyways, I had a question about the SoundTouch latency. Was there any reason why we can't use a lower (instead of 15ms) mininum latency with SoundTouch - If we can remove this requirement, I can get the latency slider inside of another function. It's sort of just a nitpick, but I'll run tests later and see if SoundTouch works with lower latency.

**Comment by turtleli** [comment]
Any progress? It should be quite simple.

**Comment by MrCK1** [comment]
Yes, I've been very busy with school unfortunately. I'll try to work on this ASAP.

**Comment by MrCK1** [comment]
@turtleli Still WIP (I had to push so I can make a new branch). 

Side note however, I can't figure out how to properly reference NullOut

**Comment by turtleli** [comment]
> Side note however, I can't figure out how to properly reference NullOut

http://en.cppreference.com/w/cpp/language/operator_member_access#Built-in_address-of_operator

**Comment by MrCK1** [comment]
@turtleli I know deep down this is easy stuff, but I can't wrap my head around this concept and how to use it in this scenario.

**Comment by turtleli** [comment]
Well, seems there's another issue - NullOut is currently only available in SndOut.cpp. An alternative would be to use FindOutputModuleById to get the module index and compare that.

But actually, looking at the code, the latency and volume sliders aren't actually related to the output module at all, so perhaps those should be not enabled/disabled depending on the selected module.

**Comment by MrCK1** [comment]
Well the sliders are only disabled for NullOut anyways, I guess we could always leave them alone. It's a pretty minor/cosmetic thing anyways.

**Comment by turtleli** [comment]
Yeah, leave them alone. As for enabling/disabling the Configure button, it might be a good idea to use willkuer's suggestion in https://github.com/PCSX2/pcsx2/pull/1963#discussion_r138656834 (Last paragraph)

**Comment by MrCK1** [comment]
I squashed my commits again, I can guarantee that my parenthesis are still wrong.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 13]
```diff
@@ -356,8 +388,8 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
             int curpos = HIWORD(wParam);
 
             switch (wmEvent) {
-                //case TB_ENDTRACK:
-                //case TB_THUMBPOSITION:
+                case TB_ENDTRACK:
+                case TB_THUMBPOSITION:
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: Explain these changes?

File: plugins/spu2-x/src/Windows/Spu2-X.rc [hunk 14]
```diff
@@ -62,15 +44,14 @@ BEGIN
     LTEXT           "Module:",IDC_STATIC,161,16,50,9,NOT WS_GROUP
     LTEXT           "(speedup!) Skips reverb effects processing, but won't sound as good in most games.",IDC_STATIC,26,60,104,36
     LTEXT           "(currently requires manual configuration via the ini file)",IDC_STATIC,162,236,146,20
-    CTEXT           "100 ms (avg)",IDC_LATENCY_LABEL,224,86,58,9
+    CTEXT           "100 ms (avg)",IDC_LATENCY_LABEL,224,84,58,9
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Spu2-X.rc]: Explain this change?

File: plugins/spu2-x/src/Windows/resource.h [hunk 15]
```diff
@@ -47,8 +46,6 @@
 #define IDC_MSG_PUBLIC_BUILD            1048
 #define IDC_XA2_TRIBLE_BUFFER           1050
 #define IDC_LABEL_VERSION_INFO          1054
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/resource.h]: This should also be removed too.

File: plugins/spu2-x/src/Windows/Spu2-X.rc [hunk 16]
```diff
@@ -24,24 +24,6 @@ LANGUAGE LANG_ENGLISH, SUBLANG_ENGLISH_US
...
-    CTEXT           "A Sound Processing Unit 2 plugin for Playstation 2 emulators.",IDC_STATIC,9,71,273,10
-    CTEXT           "For updates and news, visit the following links:",IDC_STATIC,9,98,273,10
-    CTEXT           "The SPU2-X project is derived from SPU2ghz v1.9beta released in 2008 and later modified and upgraded by the Pcsx2 Playground Team.",IDC_STATIC,25,161,242,17
-    CONTROL         "",IDC_STATIC,"Static",SS_BLACKFRAME,20,157,252,41
-    CTEXT           "Compile Date, Revision, Etc.",IDC_LABEL_VERSION_INFO,9,80,273,17
-    CTEXT           "Pcsx2 Official Git Repository @ GitHub",IDC_LINK_GOOGLECODE,9,121,273,10,SS_NOPREFIX | SS_NOTIFY
-    CTEXT           "Pcsx2 Official Website and Forums",IDC_LINK_WEBSITE,9,110,273,10,SS_NOPREFIX | SS_NOTIFY
-    CTEXT           "Brought to you by the collaborative efforts of the Pcsx2 Development Team.",IDC_STATIC,9,141,273,10
-END
-
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Spu2-X.rc]: You haven't removed everything related to the about box. Look further down in the file.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 17]
```diff
@@ -205,6 +205,35 @@ void WriteSettings()
...
+    const bool AudioExpansion =
+        mods[OutputModule] == XAudio2Out ||
+        mods[OutputModule] == PortaudioOut;
+
+    EnableWindow(GetDlgItem(window, IDC_OUTCONF), IsConfigurable);
+
+    EnableWindow(GetDlgItem(window, IDC_SPEAKERS), AudioExpansion);
+}
+
+void UpdateVolumeSlider(HWND hWnd)
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: I don't think moving this into a function really improves anything.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 18]
```diff
@@ -205,6 +205,35 @@ void WriteSettings()
...
+{
+    int wmId, wmEvent;
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: These variables are unnecessary. It seems they were copied from ConfigProc. They generate warnings that weren't there before and which should have been noticed.

File: plugins/spu2-x/src/Windows/Spu2-X.rc [hunk 19]
```diff
@@ -24,24 +24,6 @@ LANGUAGE LANG_ENGLISH, SUBLANG_ENGLISH_US
...
 //
 
-IDD_ABOUT DIALOGEX 0, 0, 291, 223
-STYLE DS_SETFONT | DS_FIXEDSYS | WS_POPUP | WS_CAPTION | WS_SYSMENU
-CAPTION "About SPU2-X"
-FONT 8, "MS Shell Dlg", 400, 0, 0x1
-BEGIN
-    DEFPUSHBUTTON   "I've Seen Enough",IDOK,91,204,111,14
-    CTEXT           "Original portions from SPU2ghz are (c) 2008 by David Quintana [gigaherz]",IDC_STATIC,9,184,273,11
-    CONTROL         114,IDC_STATIC,"Static",SS_BITMAP,45,5,203,64,WS_EX_CLIENTEDGE
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Spu2-X.rc]: Note that 114 corresponds to the IDB_SPU2X bitmap, and that it isn't used on any other dialog.

File: plugins/spu2-x/src/Windows/Spu2-X.rc (hunk 14 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Spu2-X.rc]: It moves the "latency value" up so that it's aligned with the leftmost text. On the master branch, you'll notice that the text is slightly below where it should be.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 13 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: Enabling those fixes a bug where the GUI values for volume/latency aren't updated when using the scroll wheel.

Comment (turtleli) [reply plugins/spu2-x/src/Windows/Config.cpp]: You should mention this in your commit message (actually, I'd suggest you use a separate commit for this bug fix).

TB_THUMBPOSITION should also be placed where TB_THUMBTRACK is since you don't need to send a TBM_GETPOS message for it (see https://msdn.microsoft.com/en-us/library/windows/desktop/gg430018(v=vs.85).aspx).

File: plugins/spu2-x/src/Windows/resource.h (hunk 15 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/resource.h]: OK, what about IDC_COMBO1? It's present in Lilypad and USBQemu, but it isn't used for anything at least in SPU2-X's case. Should I remove that as well?

File: plugins/spu2-x/src/Windows/Spu2-X.rc (hunk 19 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Spu2-X.rc]: Done.

File: plugins/spu2-x/src/Windows/Spu2-X.rc (hunk 16 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Spu2-X.rc]: Done.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 18 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: Done.

File: plugins/spu2-x/src/Windows/Config.cpp (hunk 17 above)
Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: Done.

File: plugins/spu2-x/src/Windows/resource.h (hunk 15 above)
Comment (turtleli) [reply plugins/spu2-x/src/Windows/resource.h]: It's up to you.

Comment (lightningterror) [reply plugins/spu2-x/src/Windows/resource.h]: If it were me I'd remove it, no need to keep useless code.

**Comment by MrCK1** [comment]
I couldn't fix the merge conflict, so I'm leaving this for now. Sorry...

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 20]
```diff
@@ -255,12 +273,14 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
             swprintf_s(temp, L"%d ms (avg)", SndOutLatencyMS);
             SetWindowText(GetDlgItem(hWnd, IDC_LATENCY_LABEL), temp);
 
-            int configvol = (int)(FinalVolume * 100 + 0.5f);
-            INIT_SLIDER(IDC_VOLUME_SLIDER, 0, 100, 10, 42, 1);
+			int configvol = (int)(FinalVolume * 100 + 0.5f);
+			INIT_SLIDER(IDC_VOLUME_SLIDER, 0, 100, 10, 42, 1);
```
Comment (danilaml) [inline plugins/spu2-x/src/Windows/Config.cpp]: mixed tabs/spaces

File: plugins/spu2-x/src/Windows/Spu2-X.rc [hunk 21]
```diff
@@ -122,6 +103,7 @@ BEGIN
     CTEXT           "8  (80 ms latency)",IDC_LATENCY_LABEL,70,37,95,11
 END
 
+    //fixme: This dialog doesn't work (was it ever used in the first place?)
```
Comment (FlatOutPS2) [inline plugins/spu2-x/src/Windows/Spu2-X.rc]: Doesn't seem to need indentation here and there should be a space after the //

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 22]
```diff
@@ -205,6 +205,24 @@ void WriteSettings()
...
+    const bool AudioExpansion =
+        mods[OutputModule] == XAudio2Out ||
+        mods[OutputModule] == PortaudioOut;
+
+    EnableWindow(GetDlgItem(window, IDC_OUTCONF), IsConfigurable);
+
+    EnableWindow(GetDlgItem(window, IDC_SPEAKERS), AudioExpansion);
+}
+
+
```
Comment (FlatOutPS2) [inline plugins/spu2-x/src/Windows/Config.cpp]: One empty line will do. ;)

**Comment by MrCK1** [comment]
@turtleli How do I remove `<None Include="..\..\spu2-x.bmp" />` from the .vcprojx file without causing a merge conflict? I added it back in temporarily but it can't stay like that, correct?

**Comment by turtleli** [comment]
Rebase to latest master first.

File: plugins/spu2-x/src/Windows/SndOut_waveOut.cpp [hunk 23]
```diff
@@ -239,14 +239,12 @@ class WaveOutModule : public SndOutModule
...
                 switch (wmId) {
-                    //case TB_ENDTRACK:
-                    //case TB_THUMBPOSITION:
                     case TB_LINEUP:
                     case TB_LINEDOWN:
                     case TB_PAGEUP:
                     case TB_PAGEDOWN:
                         wmEvent = (int)SendMessage((HWND)lParam, TBM_GETPOS, 0, 0);
-                    case TB_THUMBTRACK:
+                    case TB_THUMBPOSITION:
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/SndOut_waveOut.cpp]: My suggestion was that you have both TB_THUMBTRACK and TB_THUMBPOSITION here, not just TB_THUMBPOSITION.

You should probably add TB_TOP and TB_BOTTOM above the TBM_GETPOS line too..

Comment (MrCK1) [reply plugins/spu2-x/src/Windows/SndOut_waveOut.cpp]: OK, I'll do that.

**Comment by willkuer** [comment]
cmd line:

git checkout <your branch> // move to your branch
git branch -u pcsx2/master // set that you reference all changes to the master branch of pcsx2 (that you'd like to merge your pr)
git fetch // update your local info about the remote branch and your referenced branch
git status // check if everything is clean, all changes are commited. Unstaged changes need to be staged (git add * for all or git add -p for an interactive mode), staged changes need to be committed (git commit -m "<your commit message")
git pull // if necessary, i.e. there are remote changes
git rebase -i // adjust/rebase current branch according to the referenced branch. Here you can sqash, reorder or reword commits.
git status // check that everything is fine
git log // show latest commits, end with q
// everything is fine?
git push -f // force push to remote. Overwrite all remote changes by local ones.

**Comment by MrCK1** [comment]
I'm really rusty with merging, ect. Bear with me hear as I continue to experiment :P

**Comment by turtleli** [comment]
The first two commits need fixed. There are changes in the first commit that are immediately reverted/changed in the second commit without any good reason to do so, which isn't great.

File: plugins/spu2-x/src/Windows/Config.cpp [hunk 24]
```diff
@@ -205,6 +205,23 @@ void WriteSettings()
...
+        mods[OutputModule] == PortaudioOut ||
+        mods[OutputModule] == WaveOut ||
+        mods[OutputModule] == DSoundOut;
+
+    const bool AudioExpansion =
+        mods[OutputModule] == XAudio2Out ||
+        mods[OutputModule] == PortaudioOut;
+
+    EnableWindow(GetDlgItem(window, IDC_OUTCONF), IsConfigurable);
+
```
Comment (turtleli) [inline plugins/spu2-x/src/Windows/Config.cpp]: I'd prefer having this blank line removed.

Comment (MrCK1) [reply plugins/spu2-x/src/Windows/Config.cpp]: Sure.

**Comment by MrCK1** [comment]
@turtleli This should be all set now.

**Comment by lightningterror** [comment]
Oh no more conflicts as well, nicely done.
